src/test/java/net/sf/jsqlparser/util/CNFexpression/CNFTest.java

```diff
@@ -0,0 +1,171 @@
+package net.sf.jsqlparser.util.CNFexpression;
+
+import static org.junit.Assert.*;
+
+import java.io.StringReader;
+
+import org.junit.Test;
+
+import net.sf.jsqlparser.JSQLParserException;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.parser.CCJSqlParserManager;
+import net.sf.jsqlparser.statement.select.PlainSelect;
+import net.sf.jsqlparser.statement.select.Select;
+
+/**
+ * this class is mainly used for testing whether we generate the 
+ * correct CNF form of an expression tree. We use the name of 
+ * variables that is reflected in the steps defined in the class.
+ * @author messfish
+ *
+ */
+public class CNFTest {
+
+    private CCJSqlParserManager parserManager = new CCJSqlParserManager();
+    private Expression original1;
+    private Expression original2;
+    private Expression original3;
+    private Expression original4;
+    private Expression original5;
+    // these expressions stores the original expression tree.
+    private Expression steplast1;
+    private Expression steplast2;
+    private Expression steplast3;
+    private Expression steplast4;
+    private Expression steplast5;
+    // these expression stores the last step of the expression tree.
+    private CNFConverter cnf1;
+    private CNFConverter cnf2;
+    private CNFConverter cnf3;
+    private CNFConverter cnf4;
+    private CNFConverter cnf5;
+    // generate three CNF converter objects to handle the 
+    // conversion of three expression trees.
+    
+    /**
+     * Constructor: this constructor will be served to build several
+     * original construction tree and several expected trees that
+     * will be used in those test cases.
+     */
+    public CNFTest() {
+        /* Here is the original part of the expression tree. */
+        buildOriginalExpression1();
+        buildOriginalExpression2();
+        buildOriginalExpression3();
+        buildOriginalExpression4();
+        buildOriginalExpression5();
+        steplast1 = StepLastHelper.buildFinalStep1();
+        steplast2 = StepLastHelper.buildFinalStep2();
+        steplast3 = StepLastHelper.buildFinalStep3();
+        steplast4 = StepLastHelper.buildFinalStep4();
+        steplast5 = StepLastHelper.buildFinalStep5();
+        cnf1 = new CNFConverter();
+        cnf2 = new CNFConverter();
+        cnf3 = new CNFConverter();
+        cnf4 = new CNFConverter();
+        cnf5 = new CNFConverter();
+    }
+    
+    /**
+     * this method is used to test the full progress from the original
+     * expression tree to the expression tree in CNF form. In other words,
+     * this is used to test the last case.
+     */
+    @Test
+    public void testAll() {
```

Comment:
could we split these into individual tests? I think if we had that, we could have clearer tests with a structure:
```
// Tests CNF convertion with a NOT ( a AND b)
@Test
public void test simpleNotAnd{

String statement = "SELECT * from S WHERE NOT (1.2 < 2.3 AND 3.5 = 4.6) "
Select untransformed = (Select) JSqlTestUtil.parse(statement);

String expected = "SELECT * from S WHERE OR ( NOT ( 1.2 > 2.3 ) AND 3.5 != 4.6)" 
Select transformed = CNFConverter.convert(untransformed);
assertEquals(expected, transformed.toString()); // or even cnf.getRoot().toString();
```

However from your current tests (without looking at StepLastHelper), I have no idea what the actual expected value is, I know what your input is since I can track buildOriginalExpression1/2/3/4. However, I don't see why you can't write the test method so that it calls buildOriginalExpression1() and stores it in a local variable instead of storing it as class field. 

just my 2 cents.

Also, your CNFConverter.java class has some good doc with some trees, it would be neat if the javadoc for those tests had trees like that with the current input and the expected CNF tree.